clients/drcachesim/scheduler/scheduler.cpp

```diff
@@ -698,8 +698,10 @@ scheduler_tmpl_t<RecordType, ReaderType>::advance_region_of_interest(int output_
 {
     uint64_t cur_instr = input.reader->get_instruction_ordinal();
     auto &cur_range = input.regions_of_interest[input.cur_region];
-    if (cur_range.stop_instruction != 0 && cur_instr > cur_range.stop_instruction) {
+    if (input.in_cur_region && cur_range.stop_instruction != 0 &&
+        cur_instr > cur_range.stop_instruction) {
         ++input.cur_region;
+        input.in_cur_region = false;
```

Comment:
Are ROIs mutually exclusive? Is it possible that the next ROI overlaps with the cur one, so that when we exit the cur one, we're in the next ROI and this should still be true?

Reply:
No, they can't be.  Added docs saying this and added an up-front check and test cases to the invalid param test.